### PR TITLE
Refactor(Pagination): convert Pagination instances to PF4 

### LINKED ISSUE #128: Component: Pagination to PF4
## Acceptance Criteria
- [x] Pagination components to pf4

## Assumptions and questions
- Adjustments will happen once the table displays are updated

### Related Issue
[DISCOVERY-154](https://issues.redhat.com/browse/DISCOVERY-154)

### PR DESCRIPTION
**Pull request by jenny-s51**
## What's included

...
Closes #128 

## Example

Before:
Screen Shot 2022-07-27 at 2 02 09 PM
Screen Shot 2022-07-27 at 2 02 15 PM
Screen Shot 2022-07-27 at 2 02 22 PM

After:
Screen Shot 2022-07-27 at 1 59 58 PM
Screen Shot 2022-07-27 at 2 00 12 PM
Screen Shot 2022-07-27 at 2 00 58 PM

### CODE REVIEW

**Review by cdcabrera — CHANGES_REQUESTED** [review]
@jenny-s51 
This looks like another Redux state piece we can simplify. I gave you enough to get started, there are a few pieces missing... like passing the values back to state, updating the constants file associated with `viewPaginationConstants.js` and the unit test `viewOptionsReducer.test.js`

We'll see how it goes, then move to the next set of tweaks

**Review by jenny-s51 — COMMENTED** [review]
@cdcabrera Thank you for all your awesome and helpful suggestions on this!! I think I've addressed most of the redux-related things, still a WIP and just wanted to confirm... is this what you had in mind for the bottom pagination?

Screen Shot 2022-08-02 at 2 04 01 PM

I made a couple other updates included making the `perPage` menu open upwards, and removing the `padding-bottom: 0 !important;`
from the css class since visually we'd need to keep that bottom padding there!

**Comment by cdcabrera** [comment]
@jenny-s51 heh, close, but not quite... paging needed to stay where it was to account for the infinite page scroll. We're just using the styling from bottom paging, not the location... the top styling unfortunately removes some of the features on smaller screens, limiting accessibility and general use, blame PF for that one.

Once you move the paging back to the top, with the bottom styling, while doing the updates the required updates, I'll review

**Comment by jenny-s51** [comment]
@cdcabrera found a bug while making the most recent updates -- I tested out a whole bunch of z-index changes on our css (and did a whole bunch of experimenting in the PF docs), but can't seem to get that options menu out in front of the list content! This behavior currently exists in the PF docs so I opened an issue: https://github.com/patternfly/patternfly-react/issues/7786

Definitely a weird one! Let me know what you think is the best course of action here. A potential workaround might be to have both top and bottom pagination for now?

File: src/components/viewPaginationRow/viewPaginationRow.js [hunk 1]
```diff
@@ -1,89 +1,46 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { PaginationRow, PAGINATION_VIEW } from 'patternfly-react';
+import { Pagination } from '@patternfly/react-core';
```
Comment (cdcabrera) [inline src/components/viewPaginationRow/viewPaginationRow.js:3]: required:
import the PaginationVariant and prefer using that over any "strings", ie `PaginationVariant.bottom`. saves us future bits when PF decides to alter those enums/strings

File: src/components/viewPaginationRow/viewPaginationRow.js [hunk 2]
```diff
@@ -1,89 +1,46 @@
...
         itemCount={totalCount}
         itemsStart={itemsStart}
         itemsEnd={itemsEnd}
-        onFirstPage={this.onFirstPage}
-        onLastPage={this.onLastPage}
-        onPreviousPage={this.onPreviousPage}
-        onNextPage={this.onNextPage}
-        onPageInput={this.onPageInput}
         onPerPageSelect={this.onPerPageSelect}
+        variant="bottom"
```
Comment (cdcabrera) [inline src/components/viewPaginationRow/viewPaginationRow.js:43]: required:
apply `PaginationVariant.bottom`

File: src/redux/constants/viewPaginationConstants.js [hunk 3]
```diff
@@ -1,8 +1,4 @@
-const VIEW_FIRST_PAGE = 'VIEW_FIRST_PAGE';
-const VIEW_LAST_PAGE = 'VIEW_LAST_PAGE';
-const VIEW_PREVIOUS_PAGE = 'VIEW_PREVIOUS_PAGE';
-const VIEW_NEXT_PAGE = 'VIEW_NEXT_PAGE';
-const VIEW_PAGE_NUMBER = 'VIEW_PAGE_NUMBER';
+const VIEW_PAGE = 'VIEW_PAGE';
 const SET_PER_PAGE = 'SET_PER_PAGE';
 
-export { VIEW_FIRST_PAGE, VIEW_LAST_PAGE, VIEW_PREVIOUS_PAGE, VIEW_NEXT_PAGE, VIEW_PAGE_NUMBER, SET_PER_PAGE };
+export { VIEW_PAGE, SET_PER_PAGE };
```
Comment (cdcabrera) [inline src/redux/constants/viewPaginationConstants.js:4]: required:
the new linting updates require their to be a default here... this file should be updated to reflect that
```
...
const viewPaginationTypes = {
  SET_PER_PAGE,
  VIEW_PAGE
};

export { viewPaginationTypes as default, viewPaginationTypes, SET_PER_PAGE, VIEW_PAGE };
```

File: src/redux/reducers/__tests__/__snapshots__/viewOptionsReducer.test.js.snap [hunk 4]
```diff
@@ -9,7 +9,7 @@ Object {
       "expandedItems": Array [],
```
Comment (cdcabrera) [inline src/redux/reducers/__tests__/__snapshots__/viewOptionsReducer.test.js.snap:9]: required:
the `viewOptionsReducer.test.js` needs to be updated to reflect the changes in the reducer, your snapshots are out-of-date

Screen Shot 2022-08-04 at 11 26 34 AM

**Review by cdcabrera — CHANGES_REQUESTED** [review]
@jenny-s51 ~I can not recreate the z-index issue that you're seeing, even after~

- ~multiple reinstalls, ~
- ~a clean install of just your repo and branch~
-~making sure the latest `dev` branch changes were there ~
- ~making sure the NPM packages were updated~
- ~checking other browsers~
- ~and generally making sure my tests are coming back clean~
- ~time has now been burned... ~

~I recommend you~
- ~run through a normal clean install~
- ~investigate your tests~
- ~double check your commit rebase~
- ~**pause the bug you opened on PF and acknowledge that you still need to investigate the issue** https://github.com/patternfly/patternfly-react/issues/7786~


~Your unit snapshots are showing you exactly the z-index issue, HTML sequencing... there is a high probability this is an issue with your PR, not Patternfly.~

~If you continue to have issues, we can walk through what each of is seeing as a 1:1 after you've gone through the recommendations.~

~You've got more investigation to do~

@jenny-s51 I was finally able to recreate this issue by doing another clean install, and dropping down to below `800px`ish screen-width. That being said, **this is a non-issue for this repository, specifically, because we're planning on updating the inventory list** it is important to follow process.

We're still recommending you
- make sure your tests are updated, snapshots and all
- update your branch and commits
- and still acknowledge that this may be a non-issue for this repository on the PF issue
- **Continue with doing the updates as is.** Once that inventory update goes into place refactors will be happening based on UX-designs recommendations.

In the future, to avoid confusion and wasting time, it is recommended your branch and ALL unit tests be updated before calling out issues. It makes reviewing more difficult when things are partially updated.

File: src/redux/constants/viewPaginationConstants.js (hunk 3 above)
Comment (jenny-s51) [reply src/redux/constants/viewPaginationConstants.js]: Updated, thanks!

File: src/redux/reducers/__tests__/__snapshots__/viewOptionsReducer.test.js.snap (hunk 4 above)
Comment (jenny-s51) [reply src/redux/reducers/__tests__/__snapshots__/viewOptionsReducer.test.js.snap:9]: Updated!

**Review by jenny-s51 — COMMENTED** [review]
Thank you for your feedback and suggestions @cdcabrera - moving forward I will make sure to update tests/snapshots before raising an issue.

I've proceeded with the updates as-is, but this PR will include a buggy instance of the pagination dropdown on viewports < 800px, since we're using the bottom styling at the top. Based on your comments, it seems like that's fine for the time being?

> this is a non-issue for this repository, specifically, because we're planning on updating the inventory list 

Would you mind clarifying what you mean by this? Are you referring to #106 or something else?

Also we are going against design guidelines by implementing pagination this way -- I can close the PF issue, since this one specific use case probably doesn't need to be supported.

File: src/components/scans/__tests__/__snapshots__/scans.test.js.snap [hunk 5]
```diff
@@ -154,5 +146,13 @@ exports[`Scans Component should render a non-connected component: non-connected
...
     </ListView>
   </div>
+  <ViewPaginationRow
+    currentPage={0}
+    pageSize={0}
+    selectedItems={Array []}
+    totalCount={0}
+    variant="bottom"
+    viewType="SCANS_VIEW"
+  />
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply src/components/scans/__tests__/__snapshots__/scans.test.js.snap]: I didn't see anything in the snapshots that suggested there was an issue - the z-index behavior was independent of this! Will make sure they're fully up-to-date moving forward though

**Comment by cdcabrera** [comment]
@jenny-s51 We are headed towards release points, 3 of them to be exact, 1 we've already completed. 

This is going to sound harsh, but as a general developer you do not need the full picture, that is the role.

File: src/styles/app/_views.scss [hunk 6]
```diff
@@ -281,3 +277,7 @@
 .pficon.pficon-disconnected.is-error::before {
   color: var(--pf-global--palette--red-100);
 }
+
+.pf-c-pagination.pf-m-bottom.quipucords-view__pagination {
+  padding-bottom: 0 !important;
+}
```
Comment (cdcabrera) [inline src/styles/app/_views.scss:283]: required:
reminder...
- https://github.com/quipucords/quipucords-ui/pull/129#discussion_r934949800
- https://github.com/quipucords/quipucords-ui/pull/115#discussion_r917123345
```
.quipucords-view__pagination {
  padding-bottom: 0 !important;
}
```

**Review by cdcabrera — CHANGES_REQUESTED** [review]
Minor update

After that and the unit tests are updated, if any, and short of any other issues that pop up, we'll be good

**Review by cdcabrera — APPROVED** [review]

**Review by bclarhk — APPROVED** [review]
z-index issue to be addressed with table update.
Everything else is groovy

**Pull request merged**